File: xskeleton/src/main/java/com/mcgrady/xskeleton/mvp/IView.java
package com.mcgrady.xskeleton.mvp;

import com.trello.rxlifecycle3.LifecycleTransformer;

/**
 * Created by mcgrady on 2019/4/25.
 */
public interface IView {

    <T> LifecycleTransformer<T> bindToLifecycle();

    default void showProgress() {
    }

    default void hideProgress() {
    }

    default void finish() {

    }
}

File: xskeleton-lib/src/main/java/com/mcgrady/xskeleton/base/IView.kt
package com.mcgrady.xskeleton.base

import com.trello.rxlifecycle3.LifecycleTransformer

/**
 * Created by mcgrady on 2019-08-10.
 */
interface IView {
    fun <T> bindToLifecycle(): LifecycleTransformer<T>?
    fun showProgress() {}
    fun hideProgress() {}
    fun finish() {}
}